src/components/MemoryCard.vue

```diff
@@ -1,49 +1,89 @@
 <script lang="ts">
-import { PropType } from "vue";
-import { CardContent } from "../types/DataModels";
-export default {
+import { defineComponent, PropType } from "vue";
+import { CardData, CardType } from "../types/DataModels";
+export default defineComponent({
```

Comment:
Same as above:
We use the script setup feature which is new in vue 3 (https://github.com/Gamify-IT/docs/blob/main/dev-manuals/languages/vue/styleguide.md). Would not be that much to refactor this.